contracts/handlers/GenericHandlerV1.sol

```diff
@@ -52,60 +52,77 @@ contract GenericHandlerV1 is IGenericHandler {
+          len(executeFuncSignature):    uint16   bytes  32                                                                                          -  34
+          executeFuncSignature:         bytes    bytes  34                                                                                          -  34 + len(executeFuncSignature)
+          len(executeContractAddress):  uint8    bytes  34 + len(executeFuncSignature)                                                              -  35 + len(executeFuncSignature)
+          executeContractAddress        bytes    bytes  35 + len(executeFuncSignature)                                                              -  35 + len(executeFuncSignature) + len(executeContractAddress)
+          len(executionDataDepositor):  uint8    bytes  35 + len(executeFuncSignature) + len(executeContractAddress)                                -  36 + len(executeFuncSignature) + len(executeContractAddress)
+          executionDataDepositor:       bytes    bytes  36 + len(executeFuncSignature) + len(executeContractAddress)                                -  48 + len(executeFuncSignature) + len(executeContractAddress) + len(executionDataDepositor)
+          executionData:                bytes    bytes  48 + len(executeFuncSignature) + len(executeContractAddress) + len(executionDataDepositor)  -  END
      */
     function deposit(bytes32 resourceID, address depositor, bytes calldata data) external returns (bytes memory) {
-        require(data.length > 160, "Incorrect data length");
+        require(data.length > 81, "Incorrect data length");
 
-        uint256        lenMetadata;
-        uint256        metadataDepositor;
-        bytes   memory metaData;
+        uint256        maxFee;
+        uint16         lenExecuteFuncSignature;
+        bytes   memory executeFuncSignature;
+        uint8          lenExecuteContractAddress;
+        bytes   memory executeContractAddress;
+        uint8          lenExecutionDataDepositor;
+        bytes   memory executionDataDepositor;
+        address        executionDataDepositorAddress;
 
-        lenMetadata = abi.decode(data, (uint256));
-        metaData = bytes(data[128:128 + lenMetadata]);
+        (maxFee) = abi.decode(data, (uint256));
+        lenExecuteFuncSignature           = uint16(bytes2(data[32:34]));
+        executeFuncSignature              = bytes(data[34:34 + lenExecuteFuncSignature]);
+        lenExecuteContractAddress         = uint8(bytes1(data[34 + lenExecuteFuncSignature:35 + lenExecuteFuncSignature]));
+        executeContractAddress            = bytes(data[35 + lenExecuteFuncSignature:35 + lenExecuteFuncSignature + lenExecuteContractAddress]);
+        lenExecutionDataDepositor         = uint8(bytes1(data[35 + lenExecuteFuncSignature + lenExecuteContractAddress:36 + lenExecuteFuncSignature + lenExecuteContractAddress]));
```

Comment:
I had the same dilemma, but I'm really unsure about possible other chains so I would leave it as is

Reply:
Atleast in Solidity it is always 20 bytes, and encoded as 32 bytes. And because this is Solidity implementation we can be sure it be like that. For other VMs and compilers there will be other assumptions ofcourse, though the chain agnostic format is still valid.